include/xtensor/xcontainer.hpp

```diff
@@ -715,60 +591,68 @@ namespace xt
+    template <class D, layout L>
+    inline xstrided_container<D, L>::xstrided_container(inner_shape_type&& shape, inner_strides_type&& strides) noexcept
         : base_type(), m_shape(std::move(shape)), m_strides(std::move(strides))
     {
         m_backstrides = make_sequence<inner_backstrides_type>(m_shape.size(), 0);
         adapt_strides(m_shape, m_strides, m_backstrides);
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::shape_impl() noexcept -> inner_shape_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::shape_impl() noexcept -> inner_shape_type&
     {
         return m_shape;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::shape_impl() const noexcept -> const inner_shape_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::shape_impl() const noexcept -> const inner_shape_type&
     {
         return m_shape;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::strides_impl() noexcept -> inner_strides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::strides_impl() noexcept -> inner_strides_type&
     {
         return m_strides;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::strides_impl() const noexcept -> const inner_strides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::strides_impl() const noexcept -> const inner_strides_type&
     {
         return m_strides;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::backstrides_impl() noexcept -> inner_backstrides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::backstrides_impl() noexcept -> inner_backstrides_type&
     {
         return m_backstrides;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::backstrides_impl() const noexcept -> const inner_backstrides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::backstrides_impl() const noexcept -> const inner_backstrides_type&
     {
         return m_backstrides;
     }
 
     /**
      * Reshapes the container.
      * @param shape the new shape
+     * @param force force reshaping, even if the shape stays the same (default: false)
      */
-    template <class D>
-    inline void xstrided_container<D>::reshape(const shape_type& shape)
+    template <class D, layout L>
+    inline void xstrided_container<D, L>::reshape(const shape_type& shape, bool force)
     {
-        if (shape != m_shape)
+        if (shape != m_shape || force)
         {
-            reshape(shape, layout::row_major);
+            if (L == layout::column_major)
+            {
+                reshape(shape, layout::column_major);
+            }
+            else
+            {
+                reshape(shape, layout::row_major);
+            }
         }
```

Comment:
The layout is know at compile time, so it might be better to rely on that and select the appropriated reshape with some metaprogramming tools (tag dispatching or specializations of private implementation structures)